pkg/kubectl/cmd/util/factory.go

```diff
@@ -273,18 +279,42 @@ func getServicePorts(spec api.ServiceSpec) []string {
 }
 
 type clientSwaggerSchema struct {
-	c  *client.Client
-	ec *client.ExperimentalClient
-	t  runtime.ObjectTyper
+	c        *client.Client
+	ec       *client.ExperimentalClient
+	cacheDir string
 }
 
-func getSchemaAndValidate(c *client.RESTClient, data []byte, group, version string) error {
-	schemaData, err := c.Get().
-		AbsPath("/swaggerapi", group, version).
-		Do().
-		Raw()
-	if err != nil {
-		return err
+const schemaFileName = "schema.json"
+
+type schemaClient interface {
+	Get() *client.Request
+}
+
+func getSchemaAndValidate(c schemaClient, data []byte, group, version, cacheDir string) (err error) {
+	var schemaData []byte
+	cacheFile := path.Join(cacheDir, group, version, schemaFileName)
```

Comment:
Should it cache per server? "v1" allows different things on a 1.0 and 1.1 server